polls/views.py

```diff
@@ -81,18 +84,44 @@ def post(self, request):
 
 class TokenView(LoginRequiredMixin, UserPassesTestMixin, View):
     def get(self, request):
-        admin_api_url = request.build_absolute_uri("/polls/api/v1/admin-api/")
+        admin_api = request.build_absolute_uri(reverse("admin-api"))
+        annotation_api = request.build_absolute_uri(reverse("annotation-api"))
+        lock_users_api = request.build_absolute_uri(reverse("lock-users-api"))
         token, _ = Token.objects.get_or_create(user=request.user)
-        context = {"token": token, "admin_api_url": admin_api_url}
+        context = {
+            "token": token,
+            "admin_api": admin_api,
+            "annotation_api": annotation_api,
+            "lock_users_api": lock_users_api,
+        }
         return render(request, "polls/auth_token.html", context)
 
     def test_func(self):
         return self.request.user.is_superuser
 
 
 class AdminAPIView(APIView):
-    authentication_classes = [TokenAuthentication]
-    permission_classes = [permissions.IsAuthenticated, permissions.IsAdminUser]
-
     def get(self, request):
         return Response({"data": "hello"}, status.HTTP_200_OK)
+
+
+class AnnotationListAPI(mixins.ListModelMixin, generics.GenericAPIView):
+    queryset = Annotation.objects.all()
+    serializer_class = AnnotationSerializer
+
+    def get(self, request, *args, **kwargs):
+        return self.list(request, *args, **kwargs)
+
+
+class UserLockAPIView(APIView):
+    def post(self, request):
```

Comment:
this could be a static method, so for example instead of:
`def post(self, request):`
it could be like:
`@staticmethod`
`def post(request):`

Reply:
Which is the most common idiom in django @furqanRao and @Programmer-yar ?

Reply:
I didn't get it, why it could be a static method? I haven't seen such implementation before.
@turian this is common idiom in django and in DRF
https://www.django-rest-framework.org/tutorial/3-class-based-views/
![Screenshot from 2022-09-22 21-16-07](https://user-images.githubusercontent.com/62334099/191800386-485c2a5a-00c9-4e2e-b6ea-911799705473.png)